Extension/src/LanguageServer/client.ts

```diff
@@ -1294,20 +1299,34 @@ export class DefaultClient implements Client {
         this.rootFolder = workspaceFolder;
         this.rootRealPath = this.RootPath ? (fs.existsSync(this.RootPath) ? fs.realpathSync(this.RootPath) : this.RootPath) : "";
 
-        let storagePath: string | undefined;
+        const baseStoragePath: string | undefined = util.getBaseStoragePath();
+        let legacyStoragePath: string | undefined;
+        let storagePath: string | undefined = "";
+        let workspaceHash: string = "";
+
         if (util.extensionContext) {
```

Comment:
Can this be simplified to:

```ts
        if (util.extensionContext?.storageUri?.fsPath) {
            legacyStoragePath = util.extensionContext?.storageUri?.fsPath;
            workspaceHash = path.basename(path.dirname(legacyStoragePath));
        }
```

Reply:
I think it can be further simplified

```TypeScript
this.legacyStoragePath = util.extensionContext?.storageUri?.fsPath ?? "";
if (!this.legacyStoragePath.empty()) {
    workspaceHash = path.basename(path.dirname(this.legacyStoragePath));
}
```

That's with the local variable removed and avoid multiple calls to util.extensionContext.

Reply:
I don't think this accounts for multiple workspace folders. See:

     if (workspaceFolder && vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 1) {
            legacyStoragePath = path.join(legacyStoragePath, util.getUniqueWorkspaceStorageName(workspaceFolder));
      }